File: app/src/main/java/ru/nikshlykov/englishwordsapp/ModeDao.java
package ru.nikshlykov.englishwordsapp;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ModeDao {

    // НЕОБХОДИМО УЧЕСТЬ ТО, ЧТО ОБНОВЛЯЮТСЯ ВСЕ ПОЛЯ ИЗ ENTITY.
    @Update
    int update(Mode mode);
    // НЕОБХОДИМО УЧЕСТЬ ТО, ЧТО ОБНОВЛЯЮТСЯ ВСЕ ПОЛЯ ИЗ ENTITY.????
    @Update
    int updateMultiple(List<Mode> modes);

    // Получаем все режимы.
    @Query("SELECT * FROM Modes")
    Mode[] getAllModes();

    @Query("SELECT * FROM Modes WHERE IsSelected = 1")
    Mode[] getSelectedModes();
}

File: core-db/src/main/java/ru/nikshlykov/data/database/daos/ModeDao.kt
package ru.nikshlykov.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import androidx.room.Update
import ru.nikshlykov.data.database.models.Mode

@Dao
interface ModeDao {

  @Update
  suspend fun update(modes: List<Mode>): Int

  @Query("SELECT * FROM Modes WHERE isSelected = 1")
  suspend fun getSelectedModes(): List<Mode>

  @Query("SELECT * FROM Modes")
  suspend fun getAllModes(): List<Mode>

  @Query("SELECT COUNT(*) FROM Modes as m WHERE m.IsSelected == 1")
  fun getSelectedModesCount(): LiveData<Int>
}